packages/flame/lib/src/effects/controllers/callback_controller.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:ui';
+import 'package:flame/effects.dart';
+
+/// The [callback] will be invoked once the effect's [progress] is reached.
+/// 1.0 to match `duration` or 0.0 to match when the `reverseDuration`
+/// has been reached.
+class CallbackController extends DurationEffectController {
+  CallbackController(this.callback, {required double progress})
```

Comment:
```suggestion
/// This effect controller invokes the [callback] function and then immediately finishes. Use it as part of a more complex effect to insert callbacks at certain parts of that effect.
class CallbackController extends DurationEffectController {
  /// Creates a controller that invokes the given [callback] function.
  ///
  /// The [progress] parameter specifies the progress level of the effect at the time when the callback is invoked. It is the responsibility of the user to ensure that this progress level is contiguous with respect to the progress of the overall effect.
  CallbackController(this.callback, {required double progress})
```
(will need formatting for line length).